module_macaddress.psm1

```diff
@@ -0,0 +1,102 @@
+<#
+.SYNOPSIS
+    A simple script to retrieve MAC Address vendor by supplying MAC address
+.DESCRIPTION
+    This script downloads wireshark's Vendors Mac Address List from their gitweb.
+    This script has two functions in it:
+        * Update-MacAddressVendor
+        * Get-MacAddressVendor
+
+    Update-MacAddressVendor downloads the current vendor file and places it in a given directory, 
+    determined by the $DocumentPath variable.
+
+    Get-MacAddressVendor searches through the downloaded file for the three first octets of the MAC Address, 
+    and hopefully retrieves the correct vendor of it. 
+
+.EXAMPLE
+    # Update Vendor List
+    Update-MacAddressVendor
+
+    # Search through vendor list
+    Get-MacAddressVendor -MacAddress '00-50-56-C0-00-00'
+
+    ===================RESULT===================
+    00:50:56	Vmware	VMware, Inc.
+    ============================================
+    
+
+.NOTES
+    Script written by CodeBarbarian
+#>
+
+
+
+# Incase of a proxy
+[System.Net.WebRequest]::DefaultWebProxy.Credentials = [System.Net.CredentialCache]::DefaultCredentials
+
+
+################################# Script Config ##################################
+# -------------------------- Script Dependant Variables -------------------------#
+# Document Path - Where the vendor mac addresses textfile will be located
+$DocumentPath = ("c:$($ENV:HOMEPATH)\desktop")

```

Comment:
A better way would be to use `$DocumentPath = Join-Path $env:HOME Desktop` to avoid a hard coded dependency on the C drive.

Reply:
Correct. I really don't know why I did that. Thanks for this one.